### PR TITLE
Implements urlFromFile (minor)

### PR DESCRIPTION
**Pull request by zrrrzzt**
Adds functionality to validate urls from txt- and jsonfiles

### CODE REVIEW

File: index.js
Comment (p1ho) [inline index.js:14]: Should the imported modules be hyphen-deliminated or camelCase?
Either way, if we decide on one, i'll make everything else consistent.

File: index.js
Comment (p1ho) [inline index.js]: because of the way I implemented, the following will actually work in the original version
```PowerShell
$ site-validator alheimsins.net
```
but this will fail the ```isUrl``` test because it does not have the ```http://```.

I wonder if we should change the approach to ```isFile(input)``` instead, where we check if a file exists, if not then we convert it to a url using the normalizer and try to parse that.

Note: at least on Windows, ```alheimsins.net``` is a valid folder/file name

File: lib/get-urls-from-file.js
Comment (p1ho) [inline lib/get-urls-from-file.js:8]: I think this looks fine.
Once we merge and close this pull request, I can add the xml parsing so we have one more option for sitemaps

Comment (p1ho) [reply lib/get-urls-from-file.js:8]: I can also make a ```get-urls-from-hosted-file``` and include that in ```/lib```, where if people have hosted json/txt on their webserver say at https://alheimsins.net/test-urls.txt, it will still work.

Comment (p1ho) [reply lib/get-urls-from-file.js:8]: I wonder if this will be an issue, using the ```html-validator-cli```, I ran the following:
```$ html-validator http://asdf```
and it said the page is valid.
This means if I give it a test txt file that actually does not contain any valid urls like the following
```
random.string.1
random.random.random
```
It will actually run parse each of these as if they are url and return that they are valid...

This is what I meant:
image

the protocols would be added via the normalizer

**Review by p1ho — COMMENTED** [review]
I've left some comments, I'll merge it as soon as we decide on what we should do for the next step.

File: index.js
Comment (zrrrzzt) [reply index.js:14]: depends on your taste :-)

I usually prefer hyphens in filenames and camelCase for variables. But this is your project so I follow whatever style you prefer.

If you look at all the other requires here, they are hyphen-delimited.

File: index.js
Comment (zrrrzzt) [reply index.js]: nice catch

I'm on Mac. But isFile is fine by me. I'll fix it

File: lib/get-urls-from-file.js
Comment (zrrrzzt) [reply lib/get-urls-from-file.js:8]: Hmm, strange

File an issue? I'll have a talk with the maintainer of that module ;-)

File: index.js
Comment (p1ho) [reply index.js:14]: I like the hyphen way, after this is merged, I will change the other ones. Feel free to resolve :)

File: index.js
Comment (p1ho) [reply index.js]: OK! I'll wait for an updated version! Feel free to resolve after linking to the new change.

Comment (zrrrzzt) [reply index.js]: thought I fixed it, needs more work

File: lib/get-urls-from-file.js
Comment (p1ho) [reply lib/get-urls-from-file.js:8]: ok, I think the issue is smaller than what I thought. There's nothing wrong with html-validator per se.
I tried just the following code
```JavaScript
(async () => {
  const validator = require('html-validator')
  let data = await validator({url: 'http://asdf'})
  console.log(data)
})()
```
and it outputs
```
{"url":"http://asdf","messages":[{"type":"non-document-error","subType":"io","message":"asdf: No address associated with hostname"}]}
```
This is the correct output.

This means the ```Page is valid``` thing is just a slight issue with ```html-validator-cli``` which I'll open an issue and link here

As for what needs to be done here, we have to make a decision on whether to do a preliminary url validation, or just let w3c validator api handle it. I think this depends on whether making requests to w3c validator is costly (for example, if there's an hourly quota from an IP address, I don't think there is btw)

File: index.js
Comment (p1ho) [reply index.js]: Apologies, I'll close this and review the change there.

File: lib/get-urls-from-file.js
Comment (zrrrzzt) [reply lib/get-urls-from-file.js:8]: the w3c quota is something like "fair use" if I remember correct.

**Comment by zrrrzzt** [comment]
I think this is ready for merging now

File: index.js
Comment (p1ho) [inline index.js]: I recommend using the approach of [checking whether file exist](https://stackoverflow.com/questions/4482686/check-synchronously-if-file-directory-exists-in-node-js)
I wonder if this would be better:
```JavaScript
const fs = require('fs')

function fileFound(input) {
  return fs.existsSync(input)
}

options.url = argv.url ? normalizer(arg.url) : normalizer(arg._[0])

// Sorry for breaking the style, but if they specified --file <path-to-file>
// and it's not found, it should probably exit immediately

if (argv.file) {
  if (fileFound(argv.file)) {
    options.file = argv.file
  } else {
    console.log('File not found!')
    process.exit(1)
  }
} else {
  options.file = fileFound(argv._[0]) ? argv._[0] : false
}
```

File: lib/get-urls-from-file.js
Comment (p1ho) [reply lib/get-urls-from-file.js:8]: For the scope of this pull request, let's just go with the latter then.
I think validation beforehand should be considered as another feature.
Feel free to resolve!

If we do implement validation, we could use the [head request](https://stackoverflow.com/questions/5922842/getting-http-headers-with-node-js) approach; slightly more heavy weight but it'd be accurate. However, this assumes servers implement responding to HEAD correctly. I have dealt with sites that takes 10+ seconds to respond to a HEAD request...

File: index.js
Comment (zrrrzzt) [reply index.js]: It is always nice to work with designers :-) Yes, it is a better experience to break if a non-existing file is passed in.

Comment (p1ho) [reply index.js]: Thank you :) , I just saw your new change, I think it will work too.
Just include this check, and I'll merge and close.

**Review by p1ho — APPROVED** [review]
Nice! Thank you!

**Pull request merged**
